template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/impala/ImpalaVolumeConfigProvider.java

```diff
@@ -38,10 +40,24 @@ public List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, HostgroupV
                         .mapToInt(volume -> volume.getVolumeSize())
                         .min()
                         .orElse(0);
```

Comment:
Regarding the minAttachedVolumeSize value we calculate here (starting at line 39 above), is this the min size for the attached (non-ephemeral) volumes, or for all volumes including the ephemeral volumes?

What I think we really want is to calculate the min size for the volumes that will be used for cache and scratch, which should be:

- The min size of all ephemeral volumes if they're configured for the host group
- The min size of the attached (non ephemeral volumes) if ephemeral isn't configured.

There's a similar issue with line 65 below, where we have:
MAX_IMPALA_DATA_CACHE_SIZE_IN_GB / hostGroupView.getVolumeCount()
and the volume count we use here should be either the attached volume count or the ephemeral volume count, depending on whether we're using ephemeral.

One other thing I'll mention that's not specifically related to these ephemeral changes but might allow us to simplify some of this code a bit - I don't think we actually need the MAX_IMPALA_DATA_CACHE_SIZE_IN_GB variable any more.  That was added initially at my request way back in [CB-2766](https://jira.cloudera.com/browse/CB-2766) because of some concerns we had at that time about allowing cache sizes that were too big relative to memory, but I'm no longer concerned about that and so we could probably get rid of that variable and simplify the dataCacheCapacityInBytes calculation on lines 64 and 65 to be simply half of the min volume size, where min volume size is either the min of the ephemeral or the attached depending on what we're using.

Reply:
@drorke 
```

    The min size of all ephemeral volumes if they're configured for the host group
    The min size of the attached (non ephemeral volumes) if ephemeral isn't configured.
``` 
This is what is being implemented.
For the code removal/simplification, I'll raise another JIRA - to avoid glitches between the two.